core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -441,6 +442,8 @@ public void sendMessage(ReqT message) {
     } catch (Error e) {
       stream.cancel(Status.CANCELLED.withDescription("Client sendMessage() failed with Error"));
       throw e;
+    } finally {
+      GrpcUtil.closeQuietly(messageIs);
```

Comment:
So this is great, except it breaks DelayedStream. https://github.com/grpc/grpc-java/blob/5ac9ad0/core/src/main/java/io/grpc/internal/DelayedStream.java#L215

I see two options:
1. Have the stream take ownership of the InputStream. That's not really that hard, as it'd just be a message.close() in AbstractStream, DelayedStream, and InProcessStream.
2. Have the stream call `method.streamRequest()`. That would also remove the special-casing of RetriableStream.

Reply:
InProcessTransport already has a pattern of passing ownership of InputStream:
1. https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/inprocess/InProcessTransport.java#L378
1. https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/inprocess/InProcessTransport.java#L561

So option 1 would be more consistent with what we already do today: `AbstractStream` should close, `DelayedStream` already passes on ownership, and InProcessStream already wraps the InputStream in a `SingleMessageProducer` which passes ownership to the `StreamListener` which calls close.